.github/workflows/build-wheel-macos-x86_64.yaml

```diff
@@ -13,12 +19,23 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check_if_wheel_build_required:
+    if: >-
+      ${{
+        github.event_name != 'pull_request' ||
+        github.event.action != 'labeled' ||
+        github.event.label.name == 'ci:build-wheels'
```

Comment:
newbie here.. why do you need both `github.event.action != 'labeled'` and `github.event.label.name == 'ci:build-wheels'`? Isn't the latter sufficient here?

Reply:
Unfortunately not, it's really an AND but rather an OR. The `github.event.action != 'labeled'` is there so the build will run on update to the PR that is not a label update.

The logic is as follows, run the workflow if:
- This is not a PR (eg: push to main)
- If it is a PR, then run as long as it's not a labelling update (eg: new commit pushed to PR)
- If it is a labelling update, only run if the label being attached is `ci:build-wheels`

Reply:
Trying to understand this structure as well. Before, we were always running this job with each of the existing triggers: `pr-opened`, `pr-reopened`, `pr-synchronized`, `push-to-main`. The only additional trigger in this patch is `pr-labeled`. 

Is the `if` statement there to exclude this particular job from running on the new `pr-labeled` trigger?

Reply:
The job will get triggered when any label is added to a PR, but we only want the job to actually run if a particular label is added.

Reply:
Ok so we want to run this job if the trigger was not `pr-labeled` or it was `pr-labeled` and the label is `ci:build-wheels`, got it.

This is fairly minor then, but is `github.event.action != 'labeled'` not already true on push to main? That is  `github.event_name != 'pull_request'` is superfluous? (Either way it doesn't hurt though, it's just for my comprehension.)

Reply:
I'm closing this as well to get the PR merged since no change is required here, just a confirmation of the above.

Reply:
Hey David, your understanding is correct in that `github.event_name != 'pull_request'` is superfluous. I thought of omitting it but I felt it made the intent of the code more explicit (and readable).

Reply:
Perfect, thanks 👌